### PR TITLE
Fix AzureCommentReporter not adding comments to PR

### LINKED ISSUE #4132: AzureCommentReporter fails when using central yaml pipeline repo
**Describe the bug**
The AzureCommentReporter Python script relies on the BUILD_REPOSITORY_ID environment variable to get the PR:

image

However that only works when the yaml pipeline exists in the same repo the PR was raised for.

**To Reproduce**

1. Create a build pipeline (with `AZURE_COMMENT_REPORTER=true`) in a repo separate to code repo
2. Configure branch policy in code repo for build validation (kicks off above created pipeline)
3. Raise PR in code repo
4. Error occurs
image

**Expected behavior**
Comment on PR no matter the repository the PR was raised against

**Additional context**
Should be able to use the `SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI` environment variable and use that to locate the correct PR (extracting the repo name and looking up the repo id)

### PR DESCRIPTION
**Pull request by lukelloyd1985**
Fixes #4132 

## Proposed Changes

Use `SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI` instead of `BUILD_REPOSITORY_ID` to locate the correct repository referenced in the PR

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by echoix** [comment]
@nvuillam will that new env car be passed to the docker image on azure DevOps without further changes?

@lukelloyd1985 Will existing projects on Azure DevOps stop working if they had to pass the old env var (or it is passed without any user-specific changes)?

**Comment by nvuillam** [comment]
> @nvuillam will that new env car be passed to the docker image on azure DevOps without further changes?
> 
> @lukelloyd1985 Will existing projects on Azure DevOps stop working if they had to pass the old env var (or it is passed without any user-specific changes)?

@echoix good catch, that would be indeed a breaking change ^^

@lukelloyd1985 please can you implement the following behavior:

- if SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI has a value, then use it
- otherwise keep using BUILD_REPOSITORY_ID 

Note: I think the default azure pipelines from documentation are ok, because

**Comment by lukelloyd1985** [comment]
All (SYSTEM, BUILD etc) variables are passed by the --env-file line in the pipeline as you've noted.

SYSTEM & BUILD variables are already being utilised.

I've already added an if statement to log if the variable happens to be empty (following the logic elsewhere in the script). Should that be sufficient?

It is the PR reporter so needs to reference the correct PR branch which BUILD_REPOSITORY_ID doesn't do (unless the PR and build pipeline happen to be the same repository).

**Comment by nvuillam** [comment]
@lukelloyd1985 there are many custom pipelines using MegaLinter, probably without `--env-file <(env | grep -e SYSTEM_ -e BUILD_ -e TF_ -e AGENT_)` but with a given list of variables, like in the previous versions of MegaLinter azure-pipelines default workflow, and I don't want them to crash if they have BUILD_REPOSITORY_ID and not SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI ^^

That's why I'd like your updates to have a "fallback" with just BUILD_REPOSITORY_ID if it's the onlly one available :)

File: megalinter/reporters/AzureCommentReporter.py [hunk 1]
```diff
@@ -107,14 +101,30 @@ def produce_report(self):
             )
             git_client: GitClient = connection.clients.get_git_client()
 
-            # Get repository
-            repository = git_client.get_repository(
-                pr_repo_name
-            )
+            # Get repository id
+            if SYSTEM_PULLREQUEST_SOURCEREPOSITORYURI == "":
```
Comment (nvuillam) [inline megalinter/reporters/AzureCommentReporter.py:105]: defaut return value for config.get is `None` if a default value is not provided as 3rd parameter

**Comment by nvuillam** [comment]
> How's that?

Looks good :)

I added a last comment :)

File: megalinter/reporters/AzureCommentReporter.py (hunk 1 above)
Comment (lukelloyd1985) [reply megalinter/reporters/AzureCommentReporter.py:105]: Makes sense, corrected in the config.get

**Review by nvuillam — APPROVED** [review]
Thanks a lot for the PR :)

**Pull request merged**